### PR TITLE
13.0 mig fsm sale fix invoicing

### PR DESCRIPTION
**Pull request by hparfr**
*  _prepare_invoice_line takes no parameters
* implement https://github.com/OCA/field-service/issues/628
* remove complex code about invoicing generation
* fix view about services
* depends of #19

### CODE REVIEW

**Comment by hparfr** [comment]
Hi @brian10048 can you please review this one too ? :)

**Comment by hparfr** [comment]
Hi @brian10048  can you review this pr ?

File: fieldservice_sale/views/product_template.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
         <field name="inherit_id" ref="sale.product_template_form_view_invoice_policy" />
         <field name="arch" type="xml">
             <xpath expr="//field[@name='service_type']" position="after">
-                <field name="field_service_tracking" widget="radio" />
+                <field name="field_service_tracking" widget="radio" attrs="{'invisible': [('type', '!=', 'service')]}" />
```
Comment (brian10048) [inline fieldservice_sale/views/product_template.xml:8]: @hparfr In my use case we need to create field service orders for storable type products as well as service type products.  

```suggestion
                <field name="field_service_tracking" widget="radio" />
```

**Review by brian10048 — CHANGES_REQUESTED** [review]
I support removing the _create_invoices function.  If anyone else requires to keep this functionality, it can be added back easily via separate module.

Small change in product view that I disagree with since it would break my current usage

File: fieldservice_sale/views/product_template.xml (hunk 1 above)
Comment (hparfr) [reply fieldservice_sale/views/product_template.xml:8]: @brian10048 What is your use case ?

How is it supposed to work regarding the "shipped qty" invoice policy ? What about returned products / Partial delivery (backorders) / Qty != 1 / Legal stuff about customs...

Comment (brian10048) [reply fieldservice_sale/views/product_template.xml:8]: @hparfr

We have a product such as a tree or plant.

The only way we sell the tree to a customer is by delivering the tree to the location and providing a service to install the tree at the customer location. We still need to track the stock moves of the tree from our vendor, to the worker's truck, and to the customer.  We also need the FSM order to track and provide the service of installation.

Shipped qty policy should still be working.  If the stock moves are confirmed on the linked field service order, the sale order delivery quantity is updated.

I don't really have many situations for returned products or partial delivery etc. so I do not address these issues you may have.

I'm curious how you would configure this for my scenario.  I saw another comment where you mention using product packs?

Comment (hparfr) [reply fieldservice_sale/views/product_template.xml:8]: >I'm curious how you would configure this for my scenario. I saw another comment where you mention using product packs?

Yes https://github.com/OCA/product-pack or  https://github.com/akretion/sale-configurator may achieve what you need.
They will ensure you have on the sale order : the tree (stockable product) and installation (service) and they can ensure the quantities are increased for both lines (x trees -> x installations).
Screenshot_2021-02-25 Tree installation - Odoo
Screenshot_2021-02-25 S00029 - Odoo(1)

Comment (hparfr) [reply fieldservice_sale/views/product_template.xml:8]: So what is your position with fielrd_service_tracking ?

From my point of view, by default it should be invisible when not a service.
And your need can be resolved thanks to product_packs , sale_configuratior or simply a custom module changing the view.

But if it really matters for you, to keep it always visible, I will accept your commit suggestion :dagger:  :-)

Comment (brian10048) [reply fieldservice_sale/views/product_template.xml:8]: We can move forward with this as it is and make invisible when not a service

**Pull request merged**